lib/hypothesis.rb

```diff
@@ -2,29 +2,217 @@
+  #   expectations or minitest matchers, but anything that throws an
+  #   exception will be treated as a failed assertion.
+  #
+  # A test case which satisfies all of its assumptions and assertions
+  # is *valid*. A test-case which satisfies all of its assumptions but
+  # fails one of its assertions is *failing*.
+  #
+  # A call to hypothesis does the following:
+  #
+  # 1. It tries to *generate* a failing test case.
+  # 2. If it succeeded then it will *shrink* that failing test case.
+  # 3. Finally, it will *display* the shrunk failing test case by
+  #    the error from its failing assertion, modified to show the
+  #    givens of the test case.
+  #
+  # Generation consists of randomly trying test cases until one of
+  # three things has happened:
+  #
+  # 1. It has found a failing test case. At this point it will start
+  #    *shrinking* the test case (see below).
+  # 2. It has found enough valid test cases. At this point it will
+  #    silently stop.
+  # 3. It has found so many invalid test cases that it seems unlikely
+  #    that it will find any more valid ones in a reasonable amount of
+  #    time. At this point it will either silently stop or raise
+  #    `Hypothesis::Unsatisfiable` depending on how many valid
+  #    examples it found.
+  #
+  # *Shrinking* is when Hypothesis takes a failing test case and tries
+  # to make it easier to understand. It does this by replacing the givens
+  # in the test case with smaller and simpler values. These givens will
+  # still come from the providers, and will obey all the usual constraints.
+  # In general shrinking is automatic and you shouldn't need to care
+  # about the details of it. If the test case you're shown at the end
+  # is messy or needlessly large, please file a bug explaining the problem!
+  #
+  # @param max_valid_test_cases [Integer] The maximum number of valid test
+  #   cases to run without finding a failing test case before stopping.
+  def hypothesis(max_valid_test_cases: 200, &block)
     unless World.current_engine.nil?
       raise UsageError, 'Cannot nest hypothesis calls'
     end
     begin
-      World.current_engine = Engine.new(**options)
+      World.current_engine = Engine.new(
+        max_examples: max_valid_test_cases
+      )
       World.current_engine.run(&block)
     ensure
       World.current_engine = nil
     end
   end
 
-  def given(provider = nil, &block)
+  # Supplies a value to be used in your hypothesis.
+  # @note It is invalid to call this method outside of a hypothesis block.
+  # @return [Object] A value provided by the provider argument.
+  # @param provider [Possible] A provider that specifies the possible values
+  #   to return.
+  # @param name [String, nil] An optional name to show next to the result on
+  #   failure. This can be helpful if you have a lot of givens in your
+  #   hypothesis, as it makes it easier to keep track of which is which.
+  def any(provider, name: nil, &block)
```

Comment:
Just on an idle skim through this PR, it seems weird that a parameter called `provider` takes an instance of `Possible` – and there are a bunch of other lines where you still use the old terminology.

Reply:
Whoops. Thought I caught them all, but obviously not! Will do another pass through and grep properly this time, thanks.